recipes/pupnp/all/conanfile.py

```diff
@@ -0,0 +1,109 @@
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "ipv6": True,
+        "reuseaddr": True,
+        "webserver": True,
+        "client": True,
+        "device": True,
+        "largefile": True,
+    }
+
+    _source_subfolder = "source_subfolder"
+    _autotools = None
+
+    _pupnp_libs = ["upnp", "ixml"]
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            # Note, pupnp has build instructions for Windows but they
+            # include VC 6 and require pthreads-w32 library.
+            # Someone who needs it and has possibility to build it could step in.
+            raise ConanInvalidConfiguration("Windows builds are not supported.")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extrated_dir = self.name + "-release-" + self.version
+        os.rename(extrated_dir, self._source_subfolder)
+
+    def _configure_autotools(self):
+        if not self._autotools:
+            args = [
+                "--enable-static=%s" % ("no" if self.options.shared else "yes"),
+                "--enable-shared=%s" % ("yes" if self.options.shared else "no"),
+                "--disable-samples",
+            ]
+
+            def enable_disable(opt):
+                what = "enable" if getattr(self.options, opt) else "disable"
+                return "--{}-{}".format(what, opt)
+
+            args.extend(
+                map(
+                    enable_disable,
+                    (
+                        "ipv6",
+                        "reuseaddr",
+                        "webserver",
+                        "client",
+                        "device",
+                        "largefile",
+                    ),
+                )
+            )
+
+            self.run("./bootstrap", run_environment=True, cwd=self._source_subfolder)
+
+            self._autotools = AutoToolsBuildEnvironment(self)
+            self._autotools.configure(configure_dir=self._source_subfolder, args=args)
+        return self._autotools
+
+    def build(self):
+        autotools = self._configure_autotools()
+        autotools.make()
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        autotools = self._configure_autotools()
+        autotools.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        for lib in self._pupnp_libs:
+            os.unlink(os.path.join(self.package_folder, "lib", "lib%s.la" % lib))
+
+    def package_info(self):
+        self.cpp_info.libs.extend(self._pupnp_libs)
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs.extend(["pthread"])
+            self.cpp_info.cflags.extend(["-pthread"])
+            self.cpp_info.cxxflags.extend(["-pthread"])
+        self.cpp_info.includedirs = ["include/upnp"]
```

Comment:
```suggestion
        self.cpp_info.includedirs.append(os.path.join("include", "upnp"))
```

Reply:
Thanks for review.
I'm not sure if it's a good idea to keep both paths (default is just `["include"]` iirc).

I'm don't know why upnp puts everything into a subfolder but requires it to be in the include path 🤦‍♂️

Generally the amount of customizations and deletes makes me wonder if it's a good idea to rely on `make install` or just copy all files by glob...

Reply:
This is amt Posix-only, but I agree that `path.join` is furure-proof.
However I think having a subdir is not needed anyways so here is a "clever" solution.

If someone thinks that "clever:" is not good (I have doubts myself) I can change back to moving and removing files.